lib/src/document.dart

```diff
@@ -28,13 +38,31 @@ class Document {
     this.linkResolver,
     this.imageLinkResolver,
     this.encodeHtml = true,
-  }) : extensionSet = extensionSet ?? ExtensionSet.commonMark {
-    _blockSyntaxes
-      ..addAll(blockSyntaxes ?? [])
-      ..addAll(this.extensionSet.blockSyntaxes);
-    _inlineSyntaxes
-      ..addAll(inlineSyntaxes ?? [])
-      ..addAll(this.extensionSet.inlineSyntaxes);
+    this.withDefaultBlockSyntaxes = true,
+    this.withDefaultInlineSyntaxes = true,
+  }) {
+    hasCustomInlineSyntaxes = inlineSyntaxes?.isNotEmpty == true ||
+        extensionSet?.inlineSyntaxes.isNotEmpty == true;
+
+    _blockSyntaxes.addAll(blockSyntaxes ?? []);
+    _inlineSyntaxes.addAll(inlineSyntaxes ?? []);
+
+    if (extensionSet == null) {
+      // Add blockSyntaxes of ExtensionSet.commonMark by default if
```

Comment:
I don't think this comment is necessary; it's a rewording of the two lines of code below it.